model/src/pyrenew/process/simplerandomwalk.py

```diff
@@ -12,45 +10,58 @@ class SimpleRandomWalkProcess(RandomVariable):
     """
     Class for a Markovian
     random walk with an a
-    arbitrary step distribution
+    step distribution
     """
 
     def __init__(
         self,
-        error_distribution: dist.Distribution,
+        name: str,
+        step_rv: RandomVariable,
+        init_rv: RandomVariable,
+        t_start: int = None,
+        t_unit: int = None,
     ) -> None:
         """
         Default constructor
 
         Parameters
         ----------
-        error_distribution : dist.Distribution
-            Passed to numpyro.sample.
+        name : str
+            A name for the random variable, used to
+            name sites within it in :fun :`numpyro.sample()`
+            calls.
+        step_rv : RandomVariable
+            RandomVariable representing the step distribution.
+        init_rv : RandomVariable
+            RandomVariable representing the initial value of
+            the process
+        t_start : int
+            See :class:`RandomVariable`
+        t_unit : int
+            See :class:`RandomVariable`
 
         Returns
         -------
         None
         """
-        self.error_distribution = error_distribution
+        self.name = name
+        self.step_rv = step_rv
+        self.init_rv = init_rv
+        self.t_start = t_start
+        self.t_unit = t_unit
 
     def sample(
         self,
         n_timepoints: int,
-        name: str = "randomwalk",
-        init: float = None,
         **kwargs,
     ) -> tuple:
         """
-        Samples from the randomwalk
+        Sample from the random walk.
 
         Parameters
         ----------
         n_timepoints : int
```

Comment:
`n_timepoints` feels a bit specific for a generic randomwalk.

Reply:
`n_steps`?

Reply:
Done in f23adb5